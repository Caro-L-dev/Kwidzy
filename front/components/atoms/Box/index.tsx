/**
 * Package Import
 */
import React from "react";
import classnames from "classnames";

function Box({
  bgColor,
  children,
}: {
  bgColor: string;
  children: React.ReactNode;
}) {
  return (
    <>
      <p
        className={classnames(
          `${bgColor} flex justify-center items-center tracking-widest h-[73px] p-4 font-bold text-xl rounded-2xl my-4`
        )}
      >
        {children}
      </p>
    </>
  );
}

export default Box;
