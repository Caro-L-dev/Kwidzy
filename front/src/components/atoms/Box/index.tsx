/**
 * Package Import
 */
import React from "react";
import classnames from "classnames";

/**
 * Types
 */
interface BoxProps {
  bgColor: string;
  children: React.ReactNode;
}

/**
 * Component
 */
const Box = ({ bgColor, children }: BoxProps) => {
  return (
    <>
      <p
        className={classnames(
          ` ${bgColor} flex justify-center items-center tracking-widest
            h-[73px] p-4 font-bold text-xl rounded-2xl my-4
          `
        )}
      >
        {children}
      </p>
    </>
  );
};

export default Box;
