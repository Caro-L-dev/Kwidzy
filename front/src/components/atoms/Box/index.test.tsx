/**
 * Package Import
 */
import React from "react";
import { render, screen } from "@testing-library/react";

/**
 * Local Import
 */
import Box from "./index";
import { axe } from "jest-axe";

/**
 * Tests
 */
describe("Box", () => {
  it("should renders snapshot", () => {
    const { container } = render(<Box bgColor={""} children={""} />);
    expect(container).toMatchSnapshot();
  });

  it("should not have basic accessibility violations", async () => {
    const { container } = render(<Box bgColor={""} children={""} />);
    const results = await axe(container);
    expect(results).toHaveNoViolations();
  });
});
