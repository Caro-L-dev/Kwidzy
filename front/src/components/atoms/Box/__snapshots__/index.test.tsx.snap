// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box should renders snapshot 1`] = `
<div>
  <p
    class="  flex justify-center items-center tracking-widest
            h-[73px] p-4 font-bold text-xl rounded-2xl my-4
          "
  />
</div>
`;
